### ISSUE TITLE
Can't create API keys using alerta 4.7.16

### ISSUE THREAD

**Issue by SkylePT** [guardian/alerta#190]
Request:
`curl 'http://localhost:8080/key' -H 'Content-Type: application/json;charset=UTF-8' -H 'Accept: application/json, text/plain, */*' -H 'Authorization: Bearer XXX' --data-binary '{"user":"admin@localhost","type":"read-write","text":"yes"}'`

Response:
```
<!DOCTYPE HTML PUBLIC "-//W3C//DTD HTML 3.2 Final//EN">
<title>500 Internal Server Error</title>
<h1>Internal Server Error</h1>
<p>The server encountered an internal error and was unable to complete your request.  Either the server is overloaded or there is an error in the application.</p>
```

I'm also running it on DEBUG mode but the logs don't show anything else except:
`127.0.0.1 - - [26/Apr/2016 10:33:23] "POST /key HTTP/1.1" 500 -`

I've reverted alerta-server back to version **4.7.7** and the same requests **works** there.

**Comment by satterly** [comment]
What is the value for `AUTH_REQUIRED` ? If it is `False` then API keys aren't necessary.

Generating a `500` is not great though. What would you expect the response should be? Should the API return `400 Bad Request` and explain that you can't create API keys if authentication is not enabled? Or should the API server allow API keys to be created even though they are ignored.

**Comment by satterly** [comment]
If however, your `AUTH_REQUIRED` setting is `True` and you are getting `500` errors then that is a different matter and requires an urgent fix. Let me know.

**Comment by SkylePT** [comment]
Hey @satterly! Thanks for the fast reply.

Indeed `AUTH_REQUIRED` is set to `FALSE` because I don't want users to require to authenticate to view alerts. 

I was thinking that for generating alert events an API Token was required.

Is it possible to have something like this: Require services to use an API Token to create events but having the "GET" events public so that users don't require authentication?

**Comment by satterly** [comment]
At present it's not possible to do what you want. I would suggest you set `AUTH_REQUIRED` to `True` and then create a `read-only` API key that can be shared to all clients that need it. You could call it something easy to remember like `read-only` by manually renaming an existing one in the mongo database.

**Comment by satterly** [comment]
BTW, the problem you reported is now fixed in version 4.7.17 which is available for download immediately.

**Comment by SkylePT** [comment]
@satterly thanks! Yes, the read-only key seems to be a good workaround. I can even put a reverse proxy in front that could add the header to use the api key.
